syntax = "proto3";

package objectstore;

message PullRequest {
  bytes object_id = 1;
  bytes puller_ip = 2;
}

message PullReply {
  bool ok = 1;
}

message ReduceToRequest {
  bytes reduction_id = 1;
  oneof reduction_source {
    bytes src_object_id = 2;
  }
  bytes dst_object_id = 3;
  bytes dst_address = 4;
}

message ReduceToReply {
  bool ok = 1;
}

service ObjectStore {
  rpc Pull(PullRequest) returns (PullReply);
  rpc ReduceTo(ReduceToRequest) returns (ReduceToReply);
}
